### PR TITLE
Add picture element

### LINKED ISSUE #11416: Support responsive images
This is a large piece of work that probably depends on making our image loading code better reflect the actual specification model. For more context on what responsive images are, see [the specification introduction](https://html.spec.whatwg.org/multipage/embedded-content.html#introduction-3).

Components:
- [x] [`<picture>`](https://html.spec.whatwg.org/multipage/embedded-content.html#the-picture-element)
- [x] [`<source>`](https://html.spec.whatwg.org/multipage/embedded-content.html#the-source-element)
- [x] [sizes attribute](https://html.spec.whatwg.org/multipage/embedded-content.html#attr-img-sizes) (see #10989 for prior work)
- [x] [srcset attribute](https://html.spec.whatwg.org/multipage/embedded-content.html#attr-img-srcset) (see #10829 for prior work)
- [x] [relevant mutations](https://html.spec.whatwg.org/multipage/embedded-content.html#relevant-mutations) processing
- [x] [update the source set](https://html.spec.whatwg.org/multipage/embedded-content.html#update-the-source-set) as part of the image selection process
- [x] [environment changes](https://html.spec.whatwg.org/multipage/embedded-content.html#img-environment-changes) processing

### PR DESCRIPTION
**Pull request by Rakhisharma**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [ ] These changes are part of #11416  (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19431)

### CODE REVIEW

**Comment by Rakhisharma** [comment]
@jdm are we good to go?

File: components/script/dom/webidls/HTMLPictureElement.webidl [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+ 
+// https://html.spec.whatwg.org/multipage/#htmlimageelement
```
Comment (KiChjang) [inline components/script/dom/webidls/HTMLPictureElement.webidl]: Shouldn't this link to https://html.spec.whatwg.org/multipage/#htmlpictureelement?

**Review by KiChjang — CHANGES_REQUESTED** [review]
I believe we would also need to make changes to https://github.com/servo/servo/blob/master/components/script/dom/mod.rs, adding `htmlpictureelement` to the list, and [this list](https://github.com/servo/servo/blob/cc8f9305e415fbe773d1ea899db0755cd4a9145c/components/script/dom/create.rs#L215-L358) as well.

File: components/script/dom/webidls/HTMLPictureElement.webidl (hunk 1 above)
Comment (Rakhisharma) [reply components/script/dom/webidls/HTMLPictureElement.webidl]: ah, yes this should be https://html.spec.whatwg.org/multipage/#htmlpictureelement. I will do the required changes and update. Thanks.

**Comment by Rakhisharma** [comment]
@KiChjang I am trying to add  `htmlpictureelement` to [this list](https://github.com/servo/servo/blob/cc8f9305e415fbe773d1ea899db0755cd4a9145c/components/script/dom/create.rs#L215-L358) but I a getting build error. Please see the error [here](https://gist.github.com/Rakhisharma/5155c264d15df2658b6cb99bb6b428d7). How can I resolve this? I am not able to find any reference.

**Comment by jdm** [comment]
@rakhisharma You need to add it to https://github.com/servo/html5ever/blob/master/markup5ever/local_names.txt unfortunately.

**Comment by Rakhisharma** [comment]
@jdm I added `picture` to `servo/html5ever`. In spite of adding that I am facing few build errors here. Also in  Janitor, I have the copy of `servo/servo` repo, not `servo/html5ever`- how this works(probably I want to ask how they are communicating with each other)? because I can see the same `picture` error in my janitor build(I was facing same error before adding `picture` to  `servo/html5ever`)- https://gist.github.com/Rakhisharma/2b31baece5bcb8d2fb92a520faab6c74.

**Comment by wafflespeanut** [comment]
The `Cargo.toml` wasn't updated. I talked to her in IRC.

**Comment by jdm** [comment]
./mach test-tidy says:
```
./components/script/dom/htmlpictureelement.rs:13: trailing whitespace
./components/script/dom/htmlpictureelement.rs:13: found an empty line following a {
./components/script/dom/webidls/HTMLPictureElement.webidl:4: trailing whitespace
```
From building:
```
error[E0432]: unresolved import `dom::htmlpictureelement::HTMLPictureElement`
  --> /home/travis/build/servo/servo/components/script/dom/create.rs:57:5
   |
57 | use dom::htmlpictureelement::HTMLPictureElement;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `HTMLPictureElement` in `dom::htmlpictureelement`
error[E0432]: unresolved import `dom::htmlpictureelement::HTMLPictureElement`
   --> /home/travis/build/sevo/servo/target/debug/build/script-520c6c9af5ee635e/out/InterfaceTypes.rs:131:9
    |
131 | pub use dom::htmlpictureelement::HTMLPictureElement;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `HTMLPictureElement` in `dom::htmlpictureelement`
error[E0432]: unresolved import `dom::htmlpictureelement::HTMLPictureElement`
   --> /home/travis/build/sevo/servo/target/debug/build/script-520c6c9af5ee635e/out/InterfaceTypes.rs:131:9
    |
131 | pub use dom::htmlpictureelement::HTMLPictureElement;
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no `HTMLPictureElement` in `dom::htmlpictureelement`
error: cannot find attribute macro `dom_struct` in this scope
 --> /home/travis/build/servo/servo/components/script/dom/htmlpictureelement.rs:7:3
  |
7 | #[dom_struct]
  |   ^^^^^^^^^^
error[E0412]: cannot find type `HTMLPictureElement` in this scope
  --> /home/travis/build/servo/servo/components/script/dom/htmlpictureelement.rs:12:6
   |
12 | impl HTMLPictureElement {
   |      ^^^^^^^^^^^^^^^^^^ not found in this scope
help: you can try using the variant's enum
   |
12 | impl dom::bindings::codegen::InheritTypes::HTMLElementTypeId {
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: you can try using the variant's enum
   |
12 | impl dom::bindings::codegen::PrototypeList::ID {
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
help: you can try using the variant's enum
   |
12 | impl dom::bindings::codegen::PrototypeList::Constructor {
   |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
error: aborting due to 5 previous errors
```

File: components/script/dom/htmlpictureelement.rs [hunk 2]
```diff
@@ -0,0 +1,14 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::htmlelement::HTMLElement;
```
Comment (jdm) [inline components/script/dom/htmlpictureelement.rs]: You need `use dom_struct::dom_struct;` as well.

**Review by jdm — CHANGES_REQUESTED** [review]

**Review by jdm — CHANGES_REQUESTED** [review]
We've lost the changes to https://github.com/servo/servo/blob/cc8f9305e415fbe773d1ea899db0755cd4a9145c/components/script/dom/create.rs#L215-L358 that were previously requested.

**Comment by jdm** [comment]
```
  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] HTMLPictureElement interface: existence and properties of interface object

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] HTMLPictureElement interface object length

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] HTMLPictureElement interface object name

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] HTMLPictureElement interface: existence and properties of interface prototype object

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] HTMLPictureElement interface: existence and properties of interface prototype object's "constructor" property

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] HTMLPictureElement must be primary interface of document.createElement("picture")

  ▶ Unexpected subtest result in /html/dom/interfaces.html:
  └ PASS [expected FAIL] Stringification of document.createElement("picture")
```

**Pull request merged**